### PR TITLE
Review Assignment 3

### PR DESCRIPTION
**Pull request by himanshu8088**
Please recommend and review my solution

### CODE REVIEW

File: Tavisca.Training2017.HandsOn.Assignment3/Tavisca.Training2017.HandsOn.Assignment3/Address.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        public String Name { get; set; }
+        public String City { get; set; }
+        public String Street { get; set; }
+        public String Pin { get; set; }
+        public String HouseNo { get; set; }
+
+        public override string ToString()
+        {
+            return "Mailing Address:\n"+Name+",\n"+HouseNo+",\n"+Street+ ",\n"+City+ ",\n" + Pin;
```
Comment (pankajkawale) [inline Tavisca.Training2017.HandsOn.Assignment3/Tavisca.Training2017.HandsOn.Assignment3/Address.cs]: Strings are immutable. Each concatenation operation would create a different string. string.Format is preferred

File: Tavisca.Training2017.HandsOn.Assignment3/Tavisca.Training2017.HandsOn.Assignment3/Address.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                                                        Problem Statement 1.
+    Write a program to display user’s complete mailing address. Accept user’s name, city, street, pin and house no. and store it in a variable and display it.
+    */
+    class Address
+    {
+        public String Name { get; set; }
+        public String City { get; set; }
+        public String Street { get; set; }
+        public String Pin { get; set; }
+        public String HouseNo { get; set; }
```
Comment (pankajkawale) [inline Tavisca.Training2017.HandsOn.Assignment3/Tavisca.Training2017.HandsOn.Assignment3/Address.cs]: HouseNumber would be a better name

File: Tavisca.Training2017.HandsOn.Assignment3/Tavisca.Training2017.HandsOn.Assignment3/Solution1.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            mailingAddress.Pin = InputType<String>("Pin:");
+            mailingAddress.HouseNo = InputType<String>("House No.:");
+        }
+
+        public void DisplayMailingAddress()
+        {
+            InputType<String>(mailingAddress.ToString());
+        }
+
+        private string InputType<T>(T type)
```
Comment (pankajkawale) [inline Tavisca.Training2017.HandsOn.Assignment3/Tavisca.Training2017.HandsOn.Assignment3/Solution1.cs]: Unnecessary / incorrect use of generics. Generics are used when you would want to handle different types with same code, not when you want to handle different strings.

File: Tavisca.Training2017.HandsOn.Assignment3/Tavisca.Training2017.HandsOn.Assignment3/Address.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+        public override string ToString()
```
Comment (pankajkawale) [inline Tavisca.Training2017.HandsOn.Assignment3/Tavisca.Training2017.HandsOn.Assignment3/Address.cs]: Good use of ToString method

File: Tavisca.Training2017.HandsOn.Assignment3/Tavisca.Training2017.HandsOn.Assignment3/Program.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Tavisca.Training2017.HandsOn.Assignment3
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Solution1 sol1 = new Solution1();
```
Comment (pankajkawale) [inline Tavisca.Training2017.HandsOn.Assignment3/Tavisca.Training2017.HandsOn.Assignment3/Program.cs]: Poor class name

File: Tavisca.Training2017.HandsOn.Assignment3/Tavisca.Training2017.HandsOn.Assignment3/Solution1.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            mailingAddress.Name = InputType<String>("Enter User Name:");
+            mailingAddress.City = InputType<String>("Enter City:");
+            mailingAddress.Street = InputType<String>("Enter Street:");
+            mailingAddress.Pin = InputType<String>("Pin:");
+            mailingAddress.HouseNo = InputType<String>("House No.:");
+        }
+
+        public void DisplayMailingAddress()
+        {
+            InputType<String>(mailingAddress.ToString());
```
Comment (pankajkawale) [inline Tavisca.Training2017.HandsOn.Assignment3/Tavisca.Training2017.HandsOn.Assignment3/Solution1.cs]: InputType method asks for user input which is not needed here. Code reusability should not result into unwanted flows.

**Review by pankajkawale — CHANGES_REQUESTED** [review]

File: Tavisca.Training2017.HandsOn.Assignment3/Tavisca.Training2017.HandsOn.Assignment3/Solution1.cs (hunk 6 above)
Comment (himanshu8088) [reply Tavisca.Training2017.HandsOn.Assignment3/Tavisca.Training2017.HandsOn.Assignment3/Solution1.cs]: So is it better to include it in separate class??

File: Tavisca.Training2017.HandsOn.Assignment3/Tavisca.Training2017.HandsOn.Assignment3/Solution1.cs (hunk 3 above)
Comment (himanshu8088) [reply Tavisca.Training2017.HandsOn.Assignment3/Tavisca.Training2017.HandsOn.Assignment3/Solution1.cs]: ok.

**Pull request closed without merging**